core/src/main/java/org/primefaces/extensions/component/inputplace/InputPlaceRenderer.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2011-2024 PrimeFaces Extensions
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package org.primefaces.extensions.component.inputplace;
+
+import java.io.IOException;
+
+import javax.faces.component.UIComponent;
+import javax.faces.context.FacesContext;
+import javax.faces.context.ResponseWriter;
+
+import org.primefaces.renderkit.InputRenderer;
+import org.primefaces.util.ComponentUtils;
+import org.primefaces.util.HTML;
+import org.primefaces.util.LangUtils;
+import org.primefaces.util.WidgetBuilder;
+
+/**
+ * Renderer for the {@link InputPlace} component.
+ *
+ * @since 14.0.0
+ */
+public class InputPlaceRenderer extends InputRenderer {
+
+    @Override
+    public void decode(FacesContext context, UIComponent component) {
+        InputPlace inputPlace = (InputPlace) component;
+
+        if (!shouldDecode(inputPlace)) {
+            return;
+        }
+
+        decodeBehaviors(context, inputPlace);
+
+        String clientId = inputPlace.getClientId(context);
+        String submittedValue = context.getExternalContext().getRequestParameterMap().get(clientId);
+
+        if (submittedValue != null) {
+            int maxlength = inputPlace.getMaxlength();
+            if (maxlength > 0 && submittedValue.length() > maxlength) {
+                submittedValue = LangUtils.substring(submittedValue, 0, maxlength);
```

Comment:
This is kind of weird. Maybe we should log a warning here so the user can consider using a larger maximum length in their model.

Reply:
This is copied directly from InputTextRenderer.java in PF?

Reply:
Ah, okay, was thinking it might have been that Google's result was too large for the field's maxlength. But I guess that's clipped in the UI / JS.